app/controllers/v0/virtual_agent/virtual_agent_claim_controller.rb

```diff
@@ -97,6 +129,54 @@ def get_updated_date(claim)
         claim.list_data['claim_phase_dates']['phase_change_date']
       end
 
+      def data_for_three_most_recent_open_comp_claims_lighthouse(claims)
+        comp_claims = three_most_recent_open_comp_claims_lighthouse claims
+
+        return [] if comp_claims.nil?
+
+        transform_claims_to_response_lighthouse(comp_claims)
+      end
+
+      def three_most_recent_open_comp_claims_lighthouse(claims)
+        claims['data']
+          .sort_by { |claim| transform_to_date(claim['attributes']['claimPhaseDates']['phaseChangeDate']) }
+          .reverse
+          .select { |claim| open_compensation_lighthouse? claim }
```

Comment:
I would do the `select` before the `sort_by` as that will make reduce the amount of time the `sort_by` will take to complete

Reply:
We addressed your concern. Please confirm.